### PR TITLE
Factor out numeric case generation code & use it in other tests

### PR DESCRIPTION
**Pull request by zoddicus**
<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
PTAL, this is ready for review

File: src/unittests/maths.spec.ts [hunk 1]
```diff
@@ -842,3 +849,241 @@ g.test('test,math,correctlyRoundedFlushToZeroOnly')
...
+
+g.test('test,math,lerp')
+  .paramsSimple<lerpCase>([
+    // Infinite Bounds
+    { min: 0.0, max: Number.POSITIVE_INFINITY, t: 0.5, result: Number.NaN },
+    { min: Number.NEGATIVE_INFINITY, max: 1.0, t: 0.5, result: Number.NaN },
+    { min: Number.NEGATIVE_INFINITY, max: Number.POSITIVE_INFINITY, t: 0.5, result: Number.NaN },
+
+    // [0.0, 1.0] cases
+    { min: 0.0, max: 1.0, t: Number.NEGATIVE_INFINITY, result: 0.0 },
```
Comment (ben-clayton) [inline src/unittests/maths.spec.ts:868]: Ohhh, this is unexpected. Most `lerp` implementations do not also clamp. I'd either remove the clamping, or rename to `clampedLerp` or something.

File: src/unittests/maths.spec.ts [hunk 2]
```diff
@@ -842,3 +849,241 @@ g.test('test,math,correctlyRoundedFlushToZeroOnly')
...
+  });
+
+/**
+ * @returns true if arrays are equal, doing element-wise comparison as needed, and considering NaNs to be equal.
+ *
+ * Depends on the correctness of diffULP, which is tested earlier in this file.
+ **/
+function compareArrayOfNumbers(got: Array<number>, expect: Array<number>): boolean {
+  return (
+    got === expect ||
```
Comment (zoddicus) [inline src/unittests/maths.spec.ts:990]: TODO: Change this to use Array.every() to make more idiomatic.

File: src/unittests/maths.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/unittests/maths.spec.ts]: Done, converted to be not clamp. Also removed the requirement for the params being ordered.

**Review by zoddicus — COMMENTED** [review]
PTAL

**Review by ben-clayton — APPROVED** [review]
Lovely stuff. Great job!

**Pull request merged**